codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/HttpBoundProtocolPayloadGenerator.kt

```diff
@@ -239,12 +309,22 @@ class HttpBoundProtocolPayloadGenerator(
         val ref = if (payloadMetadata.takesOwnership) "" else "&"
         val serializer = protocolFunctions.serializeFn(member, fnNameSuffix = "http_payload") { fnName ->
             val outputT = if (member.isStreaming(model)) {
-                symbolProvider.toSymbol(member)
+                if (fromPythonServerRuntime(member)) {
```

Comment:
Rather than having `codegen-core` be aware of the Python server, we should make wrapping the streams a configurable feature of this class. See `renderEventStreamBody` as an example.

Reply:
Thanks for pointing this out. Avoiding `codegen-core` being aware of the Python server is crucial from a design point of view. Addressed in bedfb04.

The key ideas in the above commit are
- `HttpBoundProtocolPayloadGenerator` now takes additional parameter `StreamPayloadSerializer` containing two callbacks. This parameter allows client/server to pass in how they want to specify the type of serialized payload stream and how they want to wrap (or not wrap) in a new-type wrapper enabling `futures_core::stream::Stream` trait.
- For `StreamPayloadSerializer`, the server side needs to further distinguish between regular server-codegen and python server-codegen. We use `ServerHttpBoundProtocolCustomization` to make that happen; `ServerProtocolLoader.kt` and `PythonServerProtocolLoader.kt` each defines the appropriate customization.